pkl/decode_struct.go

```diff
@@ -299,7 +299,12 @@ func parseStructOpts(field *reflect.StructField) structFieldOpts {
 	return ret
 }
 
+var structFieldsCache = make(map[reflect.Type]map[string]structField)
```

Comment:
This isn't thread-safe, and also allocates structs that never get cleaned up, which isn't ideal.

What's the motivation behind this change? I'd be open to this change if we had some data that shows that this provides a performance improvement.

Reply:
Sorry, my mistake. I didn’t consider the thread safety and memory cleanup aspects. Thank you for pointing that out